src/js/memory/solverMemory.ts

```diff
@@ -33,10 +32,11 @@ export function memory_solve(e: HTMLInputElement): void {
 }
 
 function handle_inputs(element: ElementData) {
-	let row = getById(`memory_stage_${element.stage}`);
-	if (row.getAttribute("data-label")) {
+	let text = getById(`memory_label_${element.stage}`).textContent;
+
+	if (text?.startsWith("Label")) {
```

Comment:
Reading the text seems a bit crude. What about adding an attribute to the element to say what it is?
Having said that, it used to be an attribute so i think i'm misunderstanding this.

Reply:
```suggestion
	let text = getById(`memory_label_${element.stage}`)?.getAttribute("data-stage") ?? "";

	if (text == "Label") {
		data_persistance(element.stage, position, element.value);
	} else if (text == "Position") {
		data_persistance(element.stage, label, element.value);
	}

```